File: OOPS_Javascript/Polymorphism.js
function Shape() {}

Shape.prototype.draw = function() {
    console.log('Drawing a shape.');
};

function Circle() {}

Circle.prototype = Object.create(Shape.prototype);
Circle.prototype.constructor = Circle;

Circle.prototype.draw = function() {
    console.log('Drawing a circle.');
};

function Square() {}

Square.prototype = Object.create(Shape.prototype);
Square.prototype.constructor = Square;

Square.prototype.draw = function() {
    console.log('Drawing a square.');
};

function Triangle() {}

Triangle.prototype = Object.create(Shape.prototype);
Triangle.prototype.constructor = Triangle;

Triangle.prototype.draw = function() {
    console.log('Drawing a triangle.');
};

const shape1 = new Circle();
const shape2 = new Square();
const shape3 = new Triangle();

shape1.draw();
shape2.draw();
shape3.draw();

File: OOPS_Concepts/CSharp/Polymorphism.cs
using System;

public class Shape
{
    public virtual void Draw()
    {
        Console.WriteLine("Drawing a shape.");
    }
}

public class Circle : Shape
{
    public override void Draw()
    {
        Console.WriteLine("Drawing a circle.");
    }
}

public class Square : Shape
{
    public override void Draw()
    {
        Console.WriteLine("Drawing a square.");
    }
}

public class Triangle : Shape
{
    public override void Draw()
    {
        Console.WriteLine("Drawing a triangle.");
    }
}

public class Program
{
    public static void Main()
    {
        Shape shape1 = new Circle();
        Shape shape2 = new Square();
        Shape shape3 = new Triangle();

        shape1.Draw();
        shape2.Draw();
        shape3.Draw();
    }
}
